env0/resource_environment_scheduling.go

```diff
@@ -0,0 +1,128 @@
+package env0
+
+import (
+	"context"
+	"github.com/adhocore/gronx"
+	. "github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceEnvironmentScheduling() *schema.Resource {
+
+	validateCronExpression := func(i interface{}, path cty.Path) diag.Diagnostics {
+		expr := i.(string)
+		parser := gronx.New()
+		isValid := parser.IsValid(expr)
+
+		if isValid != true {
+			return diag.Diagnostics{
+				diag.Diagnostic{
+					Severity:      diag.Error,
+					Summary:       "Invalid cron expression",
+					AttributePath: path,
+				}}
+		}
+
+		return nil
+	}
+
+	return &schema.Resource{
+		CreateContext: resourceEnvironmentSchedulingCreateOrUpdate,
+		ReadContext:   resourceEnvironmentSchedulingRead,
+		UpdateContext: resourceEnvironmentSchedulingCreateOrUpdate,
+		DeleteContext: resourceEnvironmentSchedulingDelete,
+
+		Schema: map[string]*schema.Schema{
+			"environment_id": {
+				Type:        schema.TypeString,
+				Description: "the environment's id",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"destroy_cron": {
+				Type:             schema.TypeString,
+				Description:      "Cron expression for scheduled destroy of the environment",
```

Comment:
maybe add `should be different from deploy_cron`?

Reply:
Since this isn't something we state in UI, didn't feel I need to add it here. It would immediately fail once a user enters the same value...

Reply:
but the user will try to run `terraform apply` with that and we'll fail him on runtime 
i think that bad UX...

Reply:
WDYM bad UX? That's the whole purpose of schema validations in terraform - it fails on plan time

Reply:
so idk if that will fail on plan since the validation happen on the `create` which is `apply`
why not just write that in the description?
then people will see "huhh i can't have the same cron here and there" when they write there tf

Reply:
It'd fail on plan, that's how schema validations work in TF provider. You can read it in their docs and also check it yourself

Since we don't explicitly state every validation or restriction we have in our docs, it neither felt to me as time to start doing it

Reply:
I thought we were talking about the cron expression validation for a sec. Anyways, I've added it in 2cbc05e but bear in mind that we don't do that usually (for any other validation that runs on CRUD)